src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(
+  items: T[],
+  key: keyof T,
+):GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  items.forEach((item: T): void => {
+    const currentKeyValue:string = String(item[key]);
+
+    if (currentKeyValue in result) {
+      result[currentKeyValue].push(item);
+    } else {
+      result[currentKeyValue] = [item];
+    }
+  });
```

Comment:
Should work without unnecessary type descriptions. Since the iteration is already over an array of items with types T - TypeScript understands that each item will be T. There is also no need to describe the action of the forEach method, since this is a built-in method and TS knows what it does

```suggestion
  items.forEach((item) => {
    const currentKeyValue= String(item[key]);

    if (currentKeyValue in result) {
      result[currentKeyValue].push(item);
    } else {
      result[currentKeyValue] = [item];
    }
  });
```

Reply:
Thank you!
Done.